tests/cpp/test_multidevice_transformer.cpp

```diff
@@ -366,6 +441,64 @@ TEST_P(DistributedTransformerTest, MLP_Layer) {
   validate(expected_outputs, outputs);
 }
 
+TEST_P(DistributedTransformerTest, Multiheaded_Attention) {
+  auto dtype = GetParam();
+  std::unique_ptr<Fusion> fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+  auto mesh = DeviceMesh::createForNumDevices(D);
+  at::ScalarType at_dtype = data_type_to_aten(dtype);
+
+  TensorView* tvx = makeContigConcreteTensor({B * S, E}, dtype);
+  TensorView* tvw0 = makeContigConcreteTensor({D, E, 3 * E / D}, dtype);
+  TensorView* tvb0 = makeContigConcreteTensor({D, 3 * E / D}, dtype);
+  TensorView* tvw1 = makeContigConcreteTensor({D, E / D, E}, dtype);
+  TensorView* tvb1 = makeContigConcreteTensor({E}, dtype);
+
+  fusion->addInput(tvx);
+  fusion->addInput(tvw0);
+  fusion->addInput(tvb0);
+  fusion->addInput(tvw1);
+  fusion->addInput(tvb1);
+
+  auto tv_outs = mha(tvx, tvw0, tvb0, tvw1, tvb1, mesh, dtype);
+
+  for (auto tv : tv_outs) {
+    fusion->addOutput(tv);
+  }
+
+  const auto options =
+      at::TensorOptions().dtype(at_dtype).device(communicator_->device());
+  auto x = at::randn({B * S, E}, options);
+  auto w0 = at::randn({E, 3 * E}, options) * kParamScale;
+  auto b0 = at::randn({3 * E}, options) * kParamScale;
+  auto w1 = at::randn({E, E}, options) * kParamScale;
+  auto b1 = at::randn({E}, options) * kParamScale;
+
+  at::manual_seed(getATenRandomSeed());
+  auto reference_outs = reference_mha(x, w0, b0, w1, b1, at_dtype);
+  std::vector<c10::IValue> inputs = {
+      x,
+      shardTensor(w0.view({E, 3, E}), 2, mesh)
+          .view({1, E, 3 * E / D})
+          .contiguous(),
```

Comment:
Is this necessary? I remember `shardTensor` already calls `.contiguous` and `.view` doesn't change the contiguity.